### PR TITLE
Feature/api

### PR DESCRIPTION
**Pull request by ImaCherryPie**

### CODE REVIEW

File: app/Http/Controllers/Api/V1/ApiProductController.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public function store(CategoryRequest $request){
+        $product = new Product();
+        $product->name = $request->name;
+        $product->save();
+        return $product;
+    }
+
+    public function update(CategoryRequest $request, Product $product)
```
Comment (Timonm10) [inline app/Http/Controllers/Api/V1/ApiProductController.php:28]: Maybe it makes sense to create a own request for the product. (ProductRequest)
```suggestion
    public function update(ProductRequest $request, Product $product)
```

File: app/Models/product_category.php [hunk 2]
```diff
@@ -5,7 +5,7 @@
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Database\Eloquent\Model;
 
-class product_category extends Model
+class Product_Category extends Model
```
Comment (Timonm10) [inline app/Models/product_category.php:8]: Class names are UpperCamelCase
```suggestion
class ProductCategory extends Model
```

File: app/Models/product_order.php [hunk 3]
```diff
@@ -5,7 +5,7 @@
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Database\Eloquent\Model;
 
-class product_order extends Model
+class Product_Order extends Model
```
Comment (Timonm10) [inline app/Models/product_order.php:8]: Class names are UpperCamelCase
```suggestion
class ProductOrder extends Model
```

File: app/Models/Product_Category.php [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Product_Category extends Model
```
Comment (Timonm10) [inline app/Models/Product_Category.php:8]: Class names are UpperCamelCase
```suggestion
class ProductCategory extends Model
```

File: app/Models/Product_Order.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Product_Order extends Model
```
Comment (Timonm10) [inline app/Models/Product_Order.php:8]: Class names are UpperCamelCase
```suggestion
class ProductOrder extends Model
```

**Review by Timonm10 — CHANGES_REQUESTED** [review]
Good job!
Only rename some class names.

**Comment by ImaCherryPie** [comment]
All changes commited

**Review by Timonm10 — COMMENTED** [review]
LGTM

**Pull request merged**